src/shared.coffee

```diff
@@ -1166,10 +1165,6 @@ class JobCollectionBase extends Mongo.Collection
         status: newStatus
         runId: null
         after: after
-        progress:
-          completed: 0
-          total: 1
-          percent: 0
```

Comment:
I don't think this is correct for failed jobs that are set to retry.  The purpose of this code is to reset the job back to the "waiting to run" state in that case. So at a minimum, this needs to be conditional on `newStatus is 'failed'`.  

But really, even that probably doesn't achieve your goal here, because if a job fails multiple times while retrying, you will lose the progress of all but the last attempt. Perhaps you should be adding the progress info to the `err` struct so that it is preserved for each attempted run and leave this section as is.